File: src/linear_search.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

void search(const vector<string>& data, const string& value) {
    cout << "'" << value << "' 검색:" << endl;

    auto count = 0;
    for (const auto& v : data) {
        count += 1;
        if (v == value) {
            cout << count << "회 읽고 값을 찾음" << endl;
            return;
        }
    }

    cout << count << "회 수행 후 실패" << endl;
}


int main() {
    vector<string> data {
        "Hello", "My", "Name", "Is", "John Doe",
        "Something", "42", "AAAAA", "last"
    };

    search(data, "42");
    search(data, "last");
    search(data, "Hello");
    search(data, "nobody");

    return 0;
}

File: src/linear_search.py
def search(data, value):
    for i, v in enumerate(data):
        if value == v:
            print(f"{i}번 원소가 값을 가지고 있음 ({i + 1}회)")
            return v

    print(f"{value} => {len(data)}개를 다 찾아봐도 결과가 없음")
    return None


def main():
    data = ["AAA", "asdf", "qwerty", "zxcvb", "42", "hi"]
    search(data, "AAA")
    search(data, "42")
    search(data, "nope")

    pass


if __name__ == "__main__":
    main()
